skia/recorder/src/writer.cpp

```diff
@@ -0,0 +1,288 @@
+#include "writer.hpp"
+
+MovieWriter::MovieWriter(const char* _destination,
+                         int _width,
+                         int _height,
+                         int _fps)
+{
+	destinationFilename = _destination;
+	width = _width;
+	height = _height;
+	fps = _fps;
+	initialize();
+};
+
+void MovieWriter::initialize()
+{
+	// if init fails all this stuff needs cleaning up?
```

Comment:
I think you gotta make sure this gets cleaned up in the various branches yourself